sensor/common/heritage/heritage.go

```diff
@@ -0,0 +1,292 @@
+package heritage
+
+import (
+	"cmp"
+	"context"
+	"fmt"
+	"slices"
+	"sync/atomic"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+	"github.com/stackrox/rox/pkg/version"
+	v1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type PastSensor struct {
+	ContainerID   string    `json:"containerID"`
+	PodIP         string    `json:"podIP"`
+	SensorStart   time.Time `json:"sensorStart"`
+	LatestUpdate  time.Time `json:"latestUpdate"`
+	SensorVersion string    `json:"sensorVersion"`
+}
+
+// ReverseCompare compares two PastSensors to use in sorting.
+// The resulting order makes more recently updated entries are at the beginning of the slice.
+// If there are two entries with the same `LatestUpdate` (can occur only in tests), then other fields define the order.
+func (a *PastSensor) ReverseCompare(b *PastSensor) int {
+	if n := a.LatestUpdate.Compare(b.LatestUpdate); n != 0 {
+		return n * -1
+	}
+	if n := a.SensorStart.Compare(b.SensorStart); n != 0 {
+		return n * -1
+	}
+	if n := cmp.Compare(a.PodIP, b.PodIP); n != 0 {
+		return n
+	}
+	if n := cmp.Compare(a.ContainerID, b.ContainerID); n != 0 {
+		return n
+	}
+	return 0
+}
+
+func pastSensorDataString(data []*PastSensor) string {
+	str := ""
+	for i, entry := range data {
+		str = fmt.Sprintf("%s[%d]: (%s, %s) start=%s, lastUpdate=%s; ",
+			str, i, entry.ContainerID, entry.PodIP, entry.SensorStart, entry.LatestUpdate)
+	}
+	return str
+}
+
+const (
```

Comment:
Please move to the top of the file, or extract the manager to the new file.

Reply:
Here [4abbc95](https://github.com/stackrox/stackrox/pull/15176/commits/4abbc952d19d283f2dd0d03cf2e319da8ae836f0)